android/app/src/test/java/RecentMatchesViewModelTest.kt

```diff
@@ -0,0 +1,68 @@
+import com.foss.foss.feature.statsearching.recent.RecentMatchesViewModel
+import com.foss.foss.model.Match
+import com.foss.foss.repository.MatchRepository
+import com.foss.foss.util.RecentMatchesUiState
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class RecentMatchesViewModelTest {
+
+    private lateinit var matchRepository: MatchRepository
+    private lateinit var recentMatchesViewModel: RecentMatchesViewModel
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(Dispatchers.Unconfined)
+        matchRepository = mockk()
+        recentMatchesViewModel = RecentMatchesViewModel(matchRepository)
+    }
+
+    fun `최근전적 기록을 요청하면`(nickname: String) {
+        recentMatchesViewModel.fetchMatches(nickname)
+    }
+
+    @Test
+    fun `경기정보 가져오기 성공 시 UiState를 Success로 업데이트한다`() {
+        // given
+        val nickname = "사용자 이름"
+        val emptyList: List<Match> = listOf()
+
+        coEvery {
+            matchRepository.fetchMatches(nickname)
+        } returns Result.success(emptyList())
+
+        // when
+        `최근전적 기록을 요청하면`(nickname)
+        val actual = recentMatchesViewModel.uiState.value
+
+        // then
+        val expected = RecentMatchesUiState.Success(emptyList)
+        assertEquals(expected, actual)
+    }
```

Comment:
[P3]
사실 제가 의도한 바는 다음과 같이 matches의 값이 어떠한 값이 되어도 상관이 없기 때문에 이를 Fixture로 설정해 외부에서 보았을 때는 그냥 어떠한 List<Match>의 값이 있구라 라는 것을 표현하는 것을 의미했습니다.

```suggestion
    @Test
    fun `경기정보 가져오기 성공 시 UiState를 Success로 업데이트한다`() {
        // given
        val nickname = "사용자 이름"
        val matches: List<Match> = MatchFixture.matches // 어떠한 고정값
        
        coEvery {
            matchRepository.fetchMatches(nickname)
        } returns Result.success(matches)

        // when
        `최근전적 기록을 요청하면`(nickname)
        val actual = recentMatchesViewModel.uiState.value

        // then
        val expected = RecentMatchesUiState.Success(matches)
        assertEquals(expected, actual)
    }
```

Reply:
네. 근데 말씀하신 것처럼 Match의 값이 어떤 것이 들어가든 상관없는 테스트코드라 빈 리스트를 만들어서 처리했는데, 굳이 따로 Fixture를 만들어서 관리할 필요가 있나요 ? 

이 부분에 대해서 잘 몰라서 여쭤봅니다 !